azure-pipelines/pull-request-validation/build-consumers.yml

```diff
@@ -0,0 +1,196 @@
+# File: azure-pipelines\pull-request-validation\build-consumers.yml
+# Description: Test new common in ADAL / MSAL / Broker (assembleLocal and testLocalDebugUnitTest)
+# Variable: 'ENV_VSTS_MVN_ANDROIDADACCOUNTS_USERNAME' was defined in the Variables tab
+# Variable: 'mvnAccessToken' was defined in the Variables tab
+# https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+# Variable: 'commonBranchName' common branch to be used when running the pipeline manually
+name: $(date:yyyyMMdd)$(rev:.r)
+
+trigger: none
+
+variables:
+- name: robolectricSdkVersion
+  value: 28
+
+resources:
+  repositories:
+  - repository: msal
+    type: github
+    name: AzureAD/microsoft-authentication-library-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+  - repository: broker
+    type: github
+    name: AzureAD/ad-accounts-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+  - repository: adal
+    type: github
+    name: AzureAD/azure-activedirectory-library-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+
+stages:
+- stage: validateConsumers
+  displayName: Validate Consumer
+  pool:
+    name: Hosted Windows 2019 with VS2019
+  jobs:
+  # setup
+  - job: setupBranch
+    displayName: Setup branch
+    steps:
+    - checkout: none
+    - task: PowerShell@2
+      name: setvarStep
+      displayName: Set var branch
+      inputs:
+        targetType: inline
+        script: |
+          $branch = switch ( '$(Build.Reason)' )
+          {
+              PullRequest
+              {
+                '$(System.PullRequest.SourceBranch)'
+              }
+              Manual
+              {
+                '$(commonBranchName)'
+              }
+              default
+              {
+                ''
+              }
+          }
+          Write-Output "$branch"
+          echo "##vso[task.setvariable variable=commonBranch;isOutput=true]$branch"
+    - powershell: |
+        if ("$(setvarStep.commonBranch)" -eq "") {
+          Write-Output "commonBranchName is not set using default common"
```

Comment:
>commonBranchName is not set using default common [](http://example.com/codeflow?start=24&length=48)

we are not setting anything in this if block, are we setting it later?

Reply:
In this step we only inform the user the branch of common used for the validation.
if `$(setvarStep.commonBranch)` is an empty string means that we run the pipeline using an unsupported [`$(Build.Reason)`](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml) or commonBranchName was not set, the following will cause the consumers (ADAL, MSAL, Broker) use the dafault submodules

Reply:
`commonBranch` is set in the previous step 
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/c6860f3f97280704e0a6a3d0b9513e674a5c1cef/azure-pipelines/pull-request-validation/build-consumers.yml#L66

Reply:
What are all possible values for `Build.Reason`?

Reply:
https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml

![image](https://user-images.githubusercontent.com/76129899/117514295-5bf1a000-af48-11eb-8b37-b89a3dbe61c8.png)